test/ReactSetStateTestComponent.js

```diff
@@ -0,0 +1,88 @@
+function getUpdatingSetStateLifeCycleCalls() {
```

Comment:
Is the testing strategy here to implement the same mock-like implementation of a component in both JS and Dart and then verify that they produce the same order of lifecycle event dispatches?

Is there a reason not to just eliminate the JS implementation and compare the Dart output to a static list of known expectations?